api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -1131,6 +1142,49 @@ def test_update_prereg_link_info(self, app, user, preprint, url):
         assert res.status_code == 400
         assert res.json['errors'][0]['detail'] == '"maformed payload" is not a valid choice.'
 
+    def test_update_has_coi_false_with_null_conflict_statement(self, app, user, preprint, url):
+        update_payload = build_preprint_update_payload(
+            preprint._id,
+            attributes={
+                'has_coi': False,
+                'conflict_of_interest_statement': None
+            }
+        )
+
+        res = app.patch_json_api(url, update_payload, auth=user.auth)
+
+        assert res.status_code == 200
+        assert res.json['data']['attributes']['has_coi'] is False
+        assert res.json['data']['attributes']['conflict_of_interest_statement'] is None
+
+        preprint.reload()
+        assert preprint.has_coi is False
+        assert preprint.conflict_of_interest_statement is None
```

Comment:
This is great. Can you do the same for the other two assertions (links and preregistration)?

Reply:
There are 4 new tests now.